nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -157,6 +158,20 @@ Costmap2DROS::on_configure(const rclcpp_lifecycle::State & /*state*/)
 
     RCLCPP_INFO(get_logger(), "Initialized plugin \"%s\"", plugin_names_[i].c_str());
   }
+  // and costmap filters as well
+  for (unsigned int i = 0; i < filter_names_.size(); ++i) {
+    RCLCPP_INFO(get_logger(), "Using costmap filter \"%s\"", filter_names_[i].c_str());
+
+    std::shared_ptr<Layer> filter = plugin_loader_.createSharedInstance(filter_types_[i]);
+    layered_costmap_->addFilter(filter);
+
+    // TODO(mjeronimo): instead of get(), use a shared ptr
```

Comment:
remove this TODO

Reply:
Fixed in https://github.com/ros-planning/navigation2/pull/2308/commits/a9f92d8711e0bd50fe1bd3b54f2c9304c3f87b01#diff-d3369b82718255857b744f9289b4e31a34924f72fb40212604b4aa8276b857f2L168